client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -116,13 +117,57 @@ class CalypsoifyIframe extends Component< Props & ConnectedProps & ProtectedForm
 	componentDidMount() {
 		MediaStore.on( 'change', this.updateImageBlocks );
 		window.addEventListener( 'message', this.onMessage, false );
+
+		this.handleThirdPartyiFrameDomains();
 	}
 
 	componentWillUnmount() {
 		MediaStore.off( 'change', this.updateImageBlocks );
 		window.removeEventListener( 'message', this.onMessage, false );
 	}
 
+	handleThirdPartyiFrameDomains() {
+		// get the domain of the iframed editor so we can check if it is 3rd party domain.
+		// If so we want to initially redirect to that domain to set auth cookie to prevent
+		// 3rd party cookie blocking by browser security
+
+		const { hostname: iFrameDomain, protocol: iFrameProtocol } = url.parse( this.props.iframeUrl );
+		const {
+			hostname: parentDomain,
+			protocol: parentProtocol,
+			port,
+			query: parentQuery,
+		} = url.parse( window.location.href, true );
+		const parentPort = port ? `:${ port }` : '';
+		const firstPartyDomains = [ 'wordpress.com', 'calypso.live', 'calypso.localhost' ];
+		const isThirdPartyDomain = ! some( firstPartyDomains, domain => {
+			return endsWith( iFrameDomain, domain );
+		} );
```

Comment:
Let's be careful about this kind of checks. It often causes security issues when done wrong. In this case, anyone can register `evilwordpress.com` and the check will succeed for it.

Reply:
Turned out this domain checking was not needed  - just need to check if jetpack = true, so removed now